pages/pipelines/build_matrix.md.erb

```diff
@@ -0,0 +1,126 @@
+
+```yaml
+steps:
+  - label: "{{matrix}} build"
+    command: "echo '.buildkite/steps/build-binary.sh {{matrix}}'"
+    matrix:
+      - "macOS"
+      - "Linux"
+      - "Windows"
+```
+{: codeblock-file="pipeline.yml"}
+
+
+
+All jobs created by a build matrix are marked with the _Matrix_ badge in the Buildkite interface.
+
+<div class="Docs__note">
+<p class="Docs__note__heading">Matrix and Parallel steps</h3>
+<p>Matrix builds are not compatible with  <a href="/docs/tutorials/parallel-builds">parallel steps</a>. You can use both in the same build, as long as they are on separate steps.</p></div>
+
+For more complex builds, add multiple dimensions to `matrix.setup` instead of the `matrix` array. Each dimension you add is multiplied by the other dimensions, so two architectures (`matrix.setup.arch`), two operating systems (`matrix.setup.os`), and two tests (`matrix.setup.test`) create an 8 job build (2 * 2 * 2 = 8):
+
+```yaml
+steps:
+- label: "💥 Matrix Build"
+  command: "echo {{matrix.os}} {{matrix.arch}} {{matrix.test}}"
+  matrix:
+    setup:
+      arch:
+        - "amd64"
+        - "arm64"
+      os:
+        - "windows"
+        - "linux"
+      test:
+        - "A"
+        - "B"
+```
+{: codeblock-file="pipeline.yml"}
+
+
+If you're using `matrix.setup`, you can also use the `adjustments` key to change specific entries in the build matrix, or add new combinations. You can set the `skip` attribute to effectively remove them from the matrix, or `soft_fail` attributes to allow them to fail without breaking the build.
+
+```yaml
+steps:
+- label: "💥 Matrix build with adjustments"
+  command: "echo {{matrix.os}} {{matrix.arch}} {{matrix.test}}"
+  matrix:
+    setup:
+      arch:
+        - "amd64"
+        - "arm64"
+      os:
+        - "windows"
+        - "linux"
+      test:
+        - "A"
+        - "B"
+    adjustments:
+      - with:
+          os: "windows"
+          arch: "arm64"
+          test: "B"
+        soft_fail: true
+      - with:
+          os: "linux"
+          arch: "arm64"
+          test: "B"
+        skip: true
+```
+{: codeblock-file="pipeline.yml"}
+
+
+## Adding combinations to the build matrix
+
+To add an extra combination that isn't present in the `matrix.setup`, use the `adjustments` key and make sure to define all of the elements in the matrix. For example, to add a build for [Plan 9](https://en.wikipedia.org/wiki/Plan_9_from_Bell_Labs) (on `arm64`, and test suite `B`) to the previous example, use:
+
+```yaml
+    adjustments:
+      - with:
+          os: "Plan 9"
+          arch: "arm64"
+          test: "B"
+```
+{: codeblock-file="pipeline.yml"}
+
+
+This results in 9 jobs, (2 * 2 * 2 + 1 = 9).
+
+## Removing combinations from the build matrix
+
+It isn't currently possible to remove combinations using the `adjustments` key, but you can skip a combination by setting `skip: true`:
```

Comment:
Is the word "currently" here because we're planning to add support for it? Perhaps best just to leave it off until we do?

```suggestion
It isn't possible to remove combinations using the `adjustments` key, but you can skip a combination by setting `skip: true`:
```

Reply:
I think we should change this because `skip` is effectively that behaviour now; the jobs are not created